src/Data/Managers/V2/helpers/QueryControlContext.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { createContext, useContext, useState, useCallback } from "react";
+
+/**
+ * State for the QueryControlContext.
+ */
+interface QueryControlState {
+  queriesEnabled: boolean;
+}
+
+/**
+ * Interface for the Context for controlling query execution.
+ */
+interface QueryControlContextType extends QueryControlState {
+  enableQueries: () => void;
+  disableQueries: () => void;
+}
+
+/**
+ * Context for controlling query execution.
+ */
+const QueryControlContext = createContext<QueryControlContextType | undefined>(undefined);
+
+/**
+ * Provider for the QueryControlContext.
+ *
+ * @param {React.ReactNode} children - The children of the provider.
+ * @returns {React.ReactNode} The provider.
+ */
+export const QueryControlProvider: React.FC<{ children: React.ReactNode }> = ({ children }) => {
```

Comment:
This could benefit more explanation since it's a core context of the application.